monai/networks/blocks/rel_pos_embedding.py

```diff
@@ -0,0 +1,51 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Tuple
+
+import torch
+from torch import nn
+
+from monai.networks.blocks.attention_utils import add_decomposed_rel_pos
+
+
+class DecomposedRelativePosEmbedding(nn.Module):
```

Comment:
Decoupled the posembedding outside looks cleaner to me. We could also consider moving other attention components as a block such as 
https://github.com/Project-MONAI/MONAI/blob/c61c6ac2d56af08fbbfb955324b8639e266a25db/monai/networks/nets/transformer.py#L116
https://github.com/Project-MONAI/MONAI/blob/c61c6ac2d56af08fbbfb955324b8639e266a25db/monai/networks/nets/transformer.py#L138